src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/FileSystemApiTest.cs

```diff
@@ -0,0 +1,992 @@
+
+            for (var offset = 0; offset < text.Length; ++offset)
+            {
+                for (var length = text.Length + 1; 0 < length; --length)
+                {
+                    using (var data = await ipfs.UnixFsApi.ReadFileAsync(node.Id, offset, length))
+                    using (var reader = new StreamReader(data))
+                    {
+                        var readData = reader.ReadToEnd();
+                        Assert.Equal(text.Substring(offset, Math.Min(11 - offset, length)), readData);
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public async Task Read_ProtectedChunkedWithLength()
+        {
+            var text = "hello world";
+            var options = new AddFileOptions
+            {
+                ChunkSize = 3,
+                ProtectionKey = "self"
+            };
+            var node = await ipfs.UnixFsApi.AddTextAsync(text, options);
+
+            for (var offset = 0; offset < text.Length; ++offset)
+            {
+                for (var length = text.Length + 1; 0 < length; --length)
+                {
+                    using (var data = await ipfs.UnixFsApi.ReadFileAsync(node.Id, offset, length))
+                    using (var reader = new StreamReader(data))
+                    {
+                        var readData = reader.ReadToEnd();
+                        Assert.Equal(text.Substring(offset, Math.Min(11 - offset, length)), readData);
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public async Task Read_ProtectedMissingKey()
+        {
+            var text = "hello world";
+            var key = await ipfs.KeyApi.CreateAsync("alice", "rsa", 512);
+            try
+            {
+                var options = new AddFileOptions {ProtectionKey = key.Name};
+                var node = await ipfs.UnixFsApi.AddTextAsync(text, options);
+                Assert.Equal(text, await ipfs.UnixFsApi.ReadAllTextAsync(node.Id));
+
+                await ipfs.KeyApi.RemoveAsync(key.Name);
+                ExceptionAssert.Throws<KeyNotFoundException>(() =>
+                {
+                    var _ = ipfs.UnixFsApi.ReadAllTextAsync(node.Id).Result;
+                });
+            }
+            finally
+            {
+                await ipfs.KeyApi.RemoveAsync(key.Name);
+            }
+        }
+
+        [Fact]
+        public async Task AddFile_WithProgress()
+        {
+            var path = Path.GetTempFileName();
+            File.WriteAllText(path, "hello world");
+            try
+            {
+                TransferProgress lastProgress = null;
+                var options = new AddFileOptions
+                {
+                    ChunkSize = 3,
+                    Progress = new Progress<TransferProgress>(t => { lastProgress = t; })
+                };
+                var result = await ipfs.UnixFsApi.AddFileAsync(path, options);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'result'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)